subprojects/core/src/main/java/org/gradle/process/internal/daemon/WorkerDaemonClient.java

```diff
@@ -13,26 +13,24 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package org.gradle.api.internal.tasks.compile.daemon;
+package org.gradle.process.internal.daemon;
 
 import org.gradle.internal.concurrent.Stoppable;
-import org.gradle.language.base.internal.compile.CompileSpec;
-import org.gradle.language.base.internal.compile.Compiler;
 
-class CompilerDaemonClient implements CompilerDaemon, Stoppable {
+class WorkerDaemonClient implements WorkerDaemon, Stoppable {
```

Comment:
It's really an adapter from the WorkerDaemon interface to the client side of WorkerDaemonProtocol (which is the underlying communication between the build process and the worker daemon process).  I think it is appropriate to call it a client because it is the implementation of the client side of the interaction.  Of course, the _actual_ client returned by WorkerDaemonManager is a proxy that wraps around (and delegates to) this class and uses WorkerDaemonClientsManager to keep track of which clients are currently idle or not.